tests/fsharp/core/asyncStackTraces/test.fsx

```diff
@@ -135,6 +135,17 @@ let asyncTop3(f) = async {
     return ()
 }
 
+module TestFailingTaskCodeGen =
```

Comment:
The reproduction was different code:

```fsharp
let z0 () =
    task {
        let w = [| 1 |] |> Array.map (fun w -> w + 1) |> Array.head
        let! x = task { return 3 }
        let finalResult = w + x + 3
        return finalResult
    }

// Throws an NRE
z0 () |> Async.AwaitTask |> Async.RunSynchronousl
```

I think this should also be added as a test case.